ctest-next/src/template.rs

```diff
@@ -82,29 +90,94 @@ impl TestTemplate {
                     c_val: helper.c_ident(constant).into(),
                     c_ty: helper.c_type(constant)?.into(),
                 };
-                const_tests.push(item)
+                self.const_tests.push(item.clone());
+                self.test_idents.push(item.test_name);
             }
         }
 
-        let mut test_idents = vec![];
-        test_idents.extend(const_cstr_tests.iter().map(|test| test.test_name.clone()));
-        test_idents.extend(const_tests.iter().map(|test| test.test_name.clone()));
+        Ok(())
+    }
 
-        Ok(Self {
-            const_cstr_tests,
-            const_tests,
-            test_idents,
-        })
+    /// Populates size and alignment tests for aliases, structs, and unions.
+    ///
+    /// It also keeps track of the names of each test.
+    fn populate_size_align_tests(
+        &mut self,
+        helper: &TranslateHelper,
+    ) -> Result<(), TranslationError> {
+        for alias in helper.ffi_items.aliases() {
+            let item = TestSizeAlign {
+                test_name: size_align_test_ident(alias.ident()),
+                id: alias.ident().into(),
+                rust_ty: alias.ident().into(),
+                c_ty: helper.c_type(alias)?.into(),
+            };
+            self.size_align_tests.push(item.clone());
+            self.test_idents.push(item.test_name);
+        }
+        for struct_ in helper.ffi_items.structs() {
+            let item = TestSizeAlign {
+                test_name: size_align_test_ident(struct_.ident()),
+                id: struct_.ident().into(),
+                rust_ty: struct_.ident().into(),
+                c_ty: helper.c_type(struct_)?.into(),
+            };
+            self.size_align_tests.push(item.clone());
+            self.test_idents.push(item.test_name);
+        }
+        for union_ in helper.ffi_items.unions() {
+            let item = TestSizeAlign {
+                test_name: size_align_test_ident(union_.ident()),
+                id: union_.ident().into(),
+                rust_ty: union_.ident().into(),
+                c_ty: helper.c_type(union_)?.into(),
+            };
+            self.size_align_tests.push(item.clone());
+            self.test_idents.push(item.test_name);
+        }
+
+        Ok(())
+    }
+
+    /// Populates signededness tests for aliases.
+    ///
+    /// It also keeps track of the names of each test.
+    fn populate_signededness_tests(
+        &mut self,
+        helper: &TranslateHelper,
+    ) -> Result<(), TranslationError> {
+        for alias in helper.ffi_items.aliases() {
+            // && skip_signededness
+            let should_skip_signededness_test = helper
+                .generator
+                .skip_signededness
+                .as_ref()
+                .is_some_and(|skip| skip(alias.ident()));
+            if helper.translator.is_signed(helper.ffi_items, &alias.ty)
+                && !should_skip_signededness_test
```

Comment:
Early continue to unnest a level

```suggestion
            if !helper.translator.is_signed(helper.ffi_items, &alias.ty) || should_skip_signededness_test {
                continue;
            }
```